Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m

```diff
@@ -226,9 +226,18 @@ - (void)loadForDisplay
 
     DDLogError(@"%p loadForDisplay: %@", self, NSStringForOWSMessageCellType(self.cellType));
 
-    BOOL isIncoming = self.isIncoming;
-    JSQMessagesBubbleImage *bubbleImageData
-        = isIncoming ? [self.bubbleFactory incoming] : [self.bubbleFactory outgoing];
+    JSQMessagesBubbleImage *bubbleImageData;
+    if ([self.viewItem.interaction isKindOfClass:[TSOutgoingMessage class]]) {
+        TSOutgoingMessage *outgoingMessage = (TSOutgoingMessage *)self.viewItem.interaction;
+        if (outgoingMessage.messageState == TSOutgoingMessageStateUnsent) {
+            bubbleImageData = self.bubbleFactory.outgoingFailed;
+        } else {
+            bubbleImageData = self.bubbleFactory.outgoing;
```

Comment:
bubbleImageData isn't used in a number of cases, like images.  See `replaceBubbleWithView:` for a complete list of cases.  We need to do something similar for these cases too right?

Reply:
Historically we haven't done more than print the "failed to send" footer for media. I agree that we should do more to indicate failed attachments - probably something like Signal-Android.